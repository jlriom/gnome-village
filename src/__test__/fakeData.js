export const category = {
    category: "category",
    count: 0
}

export const habitantsByHairColor = [
    {
        hair_color: 'Black',
        count: 256
    },
    {
        hair_color: 'Gray',
        count: 258
    },
    {
        hair_color: 'Green',
        count: 267
    },
    {
        hair_color: 'Pink',
        count: 285
    },
    {
        hair_color: 'Red',
        count: 273
    }
];

export const habitantsByProfession = [
    {
        profession: 'Baker',
        count: 214
    },
    {
        profession: 'Blacksmith',
        count: 180
    },
    {
        profession: 'Brewer',
        count: 200
    },
    {
        profession: 'Butcher',
        count: 210
    },
    {
        profession: 'Carpenter',
        count: 216
    },
    {
        profession: 'Cook',
        count: 190
    },
    {
        profession: 'Farmer',
        count: 205
    },
    {
        profession: 'Gemcutter',
        count: 186
    },
    {
        profession: 'Leatherworker',
        count: 185
    },
    {
        profession: 'Marble Carver',
        count: 190
    },
    {
        profession: 'Mason',
        count: 214
    },
    {
        profession: 'Mechanic',
        count: 182
    },
    {
        profession: 'Medic',
        count: 187
    },
    {
        profession: 'Metalworker',
        count: 192
    },
    {
        profession: 'Miner',
        count: 201
    },
    {
        profession: 'Potter',
        count: 202
    },
    {
        profession: 'Prospector',
        count: 202
    },
    {
        profession: 'Sculptor',
        count: 182
    },
    {
        profession: 'Smelter',
        count: 203
    },
    {
        profession: 'Stonecarver',
        count: 187
    },
    {
        profession: 'Tailor',
        count: 199
    },
    {
        profession: 'Tax inspector',
        count: 192
    },
    {
        profession: ' Tinker',
        count: 202
    },
    {
        profession: 'Woodcarver',
        count: 199
    }
];

export const auth = { name: "name", isGuest: false };

export const habitant = {
    id: 362,
    name: 'Cogwitz Bittergrill',
    thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
    age: 127,
    weight: 40.074535,
    height: 93.66314,
    hair_color: 'Black',
    professions: [
        'Prospector',
        ' Tinker',
        'Baker',
        'Tailor',
        'Leatherworker',
        'Sculptor'
    ],
    friends: [
        'Whitwright Fussbuster',
        'Fizwood Bitterrocket',
        'Fizwood Felpower'
    ]
};

export const friends = [
    {
        id: 363,
        name: 'Cogwitzs friend ',
        thumbnail: 'http://www.publicdomainpictures.net/pictures/120000/nahled/white-hen.jpg',
        age: 127,
        weight: 40.074535,
        height: 93.66314,
        hair_color: 'Black',
        professions: [
            'Prospector',
            ' Tinker'
        ],
        friends: [
            'Whitwright Fussbuster'
        ]
    }
];
