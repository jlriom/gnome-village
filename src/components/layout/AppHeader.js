import React from "react";
import { Navbar, Nav } from "react-bootstrap";
import { Link } from "react-router-dom";
import LogoutLinkPopup from "../LogoutLinkPopup";
import LoginLinkPopup from "../LoginLinkPopup";

import If from "../If";

const AppHeader = ({ auth }) => {
	console.log("auth", auth);
	return (
		<Navbar bg='light' expand='lg'>
			<div className='container'>
				<Navbar.Brand as={Link} to='/'>
					Brastlewark village
				</Navbar.Brand>

				<Navbar.Toggle aria-controls='basic-navbar-nav' />
				<Navbar.Collapse id='basic-navbar-nav'>
					<Nav>
						<If condition={auth.isLoggedIn}>
							<Nav.Link as={Link} to='/dashboard'>
								Dashboard
							</Nav.Link>
							<Nav.Link as={Link} to='/brastlewark'>
								Search Habitants
							</Nav.Link>
						</If>
						<Nav.Link as={Link} to='/about'>
							About
						</Nav.Link>
					</Nav>
				</Navbar.Collapse>
				<Navbar.Collapse id='basic-navbar-nav' className='justify-content-end'>
					<Nav>
						<If condition={!auth.isLoggedIn}>
							<Nav.Link as={Link} to='/user/register'>
								Register
							</Nav.Link>
							<LoginLinkPopup />
						</If>
						<If condition={auth.isLoggedIn}>
							<If condition={!auth.isGuest}>
								<Nav.Link as={Link} to='/user/myprofile'>
									My Profile
								</Nav.Link>
								<Nav.Link as={Link} to='/user/unregister'>
									Unregister
								</Nav.Link>
							</If>
							<LogoutLinkPopup username={auth.username} />
						</If>
					</Nav>
				</Navbar.Collapse>
			</div>
		</Navbar>
	);
};

export default AppHeader;
