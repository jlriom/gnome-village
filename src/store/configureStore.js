import { createStore, applyMiddleware, compose } from "redux";
import reduxImmutableStateInvariant from "redux-immutable-state-invariant";
import thunk from "redux-thunk";
import logger from "redux-logger";

import rootReducer from "./reducers";

const middlewares = [thunk, logger, reduxImmutableStateInvariant()];

export default function configureStore(initialState) {
	const composeEnhancers =
		window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // add support for Redux dev tools

	return createStore(
		rootReducer,
		initialState,
		composeEnhancers(applyMiddleware(...middlewares))
	);
}
