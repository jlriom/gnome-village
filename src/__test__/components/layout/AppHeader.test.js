import React from "react";
import AppHeader from "../../../components/layout/AppHeader";
import renderer from "react-test-renderer";
import { auth } from "../../fakeData";
import { MemoryRouter } from 'react-router-dom';

import { Provider } from "react-redux";

import configureStore from "../../../store/configureStore";

const store = configureStore();

describe("<AppHeader />", () => {
    it("renders ok", () => {
        const tree = renderer.create(
            <Provider store={store}>
                <MemoryRouter>
                    <AppHeader auth={auth} />
                </MemoryRouter>
            </Provider>
        );
        expect(tree).toMatchSnapshot();
    });
});

