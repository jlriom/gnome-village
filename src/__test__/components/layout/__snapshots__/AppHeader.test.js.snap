// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppHeader /> renders ok 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Brastlewark village
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
         
        <a
          className="nav-link"
          data-rb-event-key={null}
          disabled={false}
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </div>
    </div>
    <div
      aria-expanded={null}
      className="justify-content-end navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key={null}
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Register
        </a>
        <a
          className="nav-link"
          data-rb-event-key={null}
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Login
        </a>
        <div />
         
         
      </div>
    </div>
  </div>
</nav>
`;
